packages/react-components/react-dialog/src/components/DialogBody/useDialogBody.ts

```diff
@@ -16,9 +16,12 @@ export const useDialogBody_unstable = (props: DialogBodyProps, ref: React.Ref<HT
     components: {
       root: 'div',
     },
-    root: getNativeElementProps(props.as ?? 'div', {
-      ref,
-      ...props,
-    }),
+    root: resolveShorthand(
+      getNativeElementProps(props.as ?? 'div', {
```

Comment:
It seems error-prone to require adding calls to `resolveShorthand` for slots that don't use shorthand (like `root` or slots that don't take children). Perhaps instead of refactoring every component's root slot, we could just have `createComponent` support having `props.children` be a render function?

Reply:
I do believe that simply start treating `root` as a proper slot would make things better patterned, and if we aim to upgrade for Option A or Option D (https://github.com/microsoft/fluentui/pull/27164), we'll have a clear signature that slots **will have** to follow, and right now `root` is not following it because we're not using `resolveShorthand` on it.